packages/schematics/src/utility/add-module-import-to-module.ts

```diff
@@ -0,0 +1,36 @@
+import { Tree, SchematicsException } from '@angular-devkit/schematics';
+import { InsertChange, Change } from '@schematics/angular/utility/change';
+import { addImportToModule } from '@schematics/angular/utility/ast-utils';
+import { getTypeScriptSourceFile } from './get-typescript-source-file';
+
+/**
+ * Import and add module to specific module path.
+ * @param host {Tree} The source tree.
+ * @param importedModuleName {String} The name of the imported module.
+ * @param importedModulePath {String} The location of the imported module.
+ * @param moduleToImportIn {String} The location of the module to import in.
+ */
+export function addModuleImportToModule(
+  host: Tree,
+  importedModuleName: string,
+  importedModulePath: string,
+  moduleToImportIn?: string
+) {
+  if (!moduleToImportIn) {
+    throw new SchematicsException(`Entry module not found.`);
+  }
+
+  const moduleSource = getTypeScriptSourceFile(host, moduleToImportIn);
+  if (!moduleSource) {
+    throw new SchematicsException(`Module not found: ${moduleToImportIn}`);
```

Comment:
Similar question for this guard. If the function arg `moduleSource` is required but not provided by the caller ~couldn't~ shouldn't we depend on `tsc` to fail with a syntax error? 

The reason for asking is that this seems to look like un-necesarry code but I've not tested it locally to make sure.

Reply:
`moduleSource` might be an empty string